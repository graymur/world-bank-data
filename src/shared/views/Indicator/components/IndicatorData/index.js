import React from 'react';
import PropTypes from 'prop-types';
import Loader from 'shared/components/Loader';
import {ResponsiveContainer, BarChart, CartesianGrid, XAxis, YAxis, Tooltip, Bar} from 'recharts';
import classnames from 'classnames';
import './indicator-data.scss';

export class IndicatorData extends React.PureComponent {
	static propTypes = {
		loading: PropTypes.bool,
		data: PropTypes.array
	};

	renderData() {
		const {data} = this.props;

		if (!data) {
			return null;
		}

		if (!data.length) {
			return (
				<div id='chart' className='indicator__data__chart _empty'>
					<h2>No data for this year.</h2>
				</div>
			);
		}

		const height = data.length * 3;

		return (
			<div id='chart' className='indicator__data__chart' style={{height: `${height}rem`}}>
				<ResponsiveContainer>
					<BarChart layout='vertical' data={data}>
						<CartesianGrid strokeDasharray='3 3'/>
						<XAxis type='number' dataKey='value'/>
						<YAxis type='category' dataKey='name' width={150}/>
						<Tooltip/>
						<Bar dataKey='value' fill='#8884d8' />
					</BarChart>
				</ResponsiveContainer>
			</div>
		);
	}

	render() {
		const {loading} = this.props;
		const classNames = classnames('indicator__data', {'loading': loading});

		return (
			<div className={classNames}>
				{loading ? <Loader/> : this.renderData()}
			</div>
		);
	}
}

export default IndicatorData;
