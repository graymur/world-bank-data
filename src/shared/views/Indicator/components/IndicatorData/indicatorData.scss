.indicator__data {
    margin-top: 3rem;

    &__chart {
        width: 100%;
        max-width: 70rem;
        margin: 0 auto;

        &._empty {
            padding-top: 2rem;
            text-align: center;
        }
    }
}