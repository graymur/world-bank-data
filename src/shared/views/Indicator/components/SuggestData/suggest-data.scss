.indicator__data__suggestions {
    text-align: center;

    &__item {
        display: inline-block;
        margin: 0 0 0 0.5rem;
    }
}