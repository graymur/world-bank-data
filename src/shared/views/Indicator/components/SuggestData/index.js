import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import './suggest-data.scss';

export class SuggestData extends React.PureComponent {
	static propTypes = {
		indicatorId: PropTypes.string,
		suggestData: PropTypes.object
	};

	render() {
		const {suggestData, indicatorId} = this.props;

		const years = Object.keys(suggestData).sort();

		if (!years.length) {
			return null;
		}

		return (
			<div className='indicator__data__suggestions'>
				Try these years instead:
				{years.map(year => (
					<Link to={`/indicators/${indicatorId}/${year}#chart`} key={year} className='indicator__data__suggestions__item'>{year}</Link>
				))}
			</div>
		);
	}
}

export default SuggestData;
