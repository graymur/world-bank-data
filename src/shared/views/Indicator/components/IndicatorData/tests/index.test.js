/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import IndicatorData from '../';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SuggestData from '../../SuggestData'

configure({adapter: new Adapter()});

const props = {
	loading: false,
	data: [{
		name: 1,
		value: 1,
	}, {
		name: 2,
		value: 2,
	}]
};

test('Renders preloader', () => {
	const thisProps = Object.assign({}, props);
	thisProps.loading = true;
	const component = renderer.create(<IndicatorData loading={true}/>);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Renders data', () => {
	const thisProps = Object.assign({}, props);
	thisProps.loading = true;
	const component = shallow(<IndicatorData {...props}/>)

	expect(component.find('.indicator__data__chart').length).toBe(1)
	expect(component.find('.loader').length).toBe(0)
});

test('Renders data suggestion component', () => {
	const thisProps = Object.assign({}, props);
	thisProps.loading = true;
	const component = shallow(<IndicatorData loading={false} indicator={{id: '1'}} currentYear={2000} data={[]}/>);

	expect(component.find('.indicator__data__chart._empty').length).toBe(1)
	expect(component.find(SuggestData).length).toBe(1)
});
