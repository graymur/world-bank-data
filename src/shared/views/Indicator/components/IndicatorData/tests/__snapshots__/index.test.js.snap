// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders preloader 1`] = `
<div
  className="indicator__data loading"
>
  <span
    className="loader"
    style={
      Object {
        "border": "10px solid #f3f3f3",
        "borderTop": "10px solid #3498db",
        "height": "50px",
        "width": "50px",
      }
    }
  />
</div>
`;
